### ISSUE TITLE
Fixed in PR #146 - newint.__divmod__(<float>) barfs with NotImplemented

### ISSUE THREAD

**Issue by posita** [PythonCharmers/python-future#144]
*UPDATE: PR #146 submitted to address this.*

--

```py
>>> from builtins import int as _int
>>> i = int(1431216000)
>>> j = _int(i)
>>> i == j
True
>>> type(i)
<type 'int'>
>>> type(j)
<class 'future.types.newint.newint'>
>>> divmod(i, 1.0)
(1431216000.0, 0.0)
>>> divmod(j, 1.0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../site-packages/future-0.14.3-py2.7.egg/future/types/newint.py", line 204, in __divmod__
    return (newint(value[0]), newint(value[1]))
TypeError: 'NotImplementedType' object has no attribute '__getitem__'
>>> import future
>>> future.__version__
'0.14.3'
```

On PyPy (2.5.1), this breaks `datetime.utcfromtimestamp`:

```py
>>>> from builtins import int as _int
>>>> i = int(1431216000)
>>>> j = _int(i)
>>>> i == j
True
>>>> type(i)
<type 'int'>
>>>> type(j)
<class 'future.types.newint.newint'>
>>>> divmod(i, 1.0)
(1431216000.0, 0.0)
>>>> divmod(j, 1.0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../site-packages/future-0.14.3-py2.7.egg/future/types/newint.py", line 204, in __divmod__
    return (newint(value[0]), newint(value[1]))
TypeError: 'NotImplemented' object is not subscriptable
>>>> import future
>>>> future.__version__
'0.14.3'
>>>> from datetime import datetime
>>>> datetime.utcfromtimestamp(i)
datetime.datetime(2015, 5, 10, 0, 0)
>>>> datetime.utcfromtimestamp(j)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../lib_pypy/datetime.py", line 1530, in utcfromtimestamp
    t, frac = divmod(t, 1.0)
  File ".../site-packages/future-0.14.3-py2.7.egg/future/types/newint.py", line 204, in __divmod__
    return (newint(value[0]), newint(value[1]))
TypeError: 'NotImplemented' object is not subscriptable
>>>> from sys import version
>>>> print(version)
2.7.9 (9c4588d731b7fe0b08669bd732c2b676cb0a8233, Apr 02 2015, 15:24:16)
[PyPy 2.5.1 with GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]
```

Curiously, this doesn't happen in cPython (session continued from above):

```py
>>> from datetime import datetime
>>> datetime.utcfromtimestamp(i)
datetime.datetime(2015, 5, 10, 0, 0)
>>> datetime.utcfromtimestamp(j)
datetime.datetime(2015, 5, 10, 0, 0)
>>> from sys import version
>>> print(version)
2.7.9 (default, Dec 12 2014, 13:52:29)
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.56)]
```

**Comment by QuLogic** [comment]
`newint` seems to derive from `long`. Have you tried the same with a native `long` instead of a native `int`?

**Comment by posita** [comment]
Using a `long` works in cPython:

```py
>>> from datetime import datetime
>>> i = 1431216000L
>>> type(i)
<type 'long'>
>>> divmod(i, 1.0)
(1431216000.0, 0.0)
>>> datetime.utcfromtimestamp(i)
datetime.datetime(2015, 5, 10, 0, 0)
```

... as well as PyPy:

```py
>>>> from datetime import datetime
>>>> i = 1431216000L
>>>> type(i)
<type 'long'>
>>>> divmod(i, 1.0)
(1431216000.0, 0.0)
>>>> datetime.utcfromtimestamp(i)
datetime.datetime(2015, 5, 10, 0, 0)
```

See also https://bitbucket.org/pypy/pypy/issue/2048/. However, despite the `datetime` oddity, `future` should probably support `newint.__divmod__(<float>)`, if it is to be a true drop-in replacement for Python 2.7's `int`.

**Comment by edschofield** [comment]
Thanks very much for finding and fixing this. Merged into the `v0.14.x` branch.
